Homework patterns.md

```diff
@@ -0,0 +1,14 @@
+Design Patterns: Is a solution to fix the common problem than can occur in software design. The design pattern can considered as a templates because aren’t complete and this allow that we can used them by different ways to solve problems.
+
+Factory pattern: This is type of design pattern in Java, in which we have an interface “factory” that can create different’s type of object depending of the classes need. By example an interface than can draw shapes is considered a factory of shapes and this interface can create shapes such as circles, triangles and rectangles. This objects are different one of other but at the same time are shapes.
+This type of design patter allow hide the “factory” to the client, because they only need use the classes to build their objects. This use the polymorphism because the classes can override the method’s factory and use as they required. 
+
```

Comment:
As part of this homework, please add a package for each one of these patterns and write examples of every one in Java.